ext/spl/spl_array.c

```diff
@@ -74,6 +74,11 @@ static inline HashTable **spl_array_get_hash_table_ptr(spl_array_object* intern)
 		return &Z_ARRVAL(intern->array);
 	} else {
 		zend_object *obj = Z_OBJ(intern->array);
+		/* Since we're directly playing with the properties table, we shall initialize the lazy object directly.
+		 * If we don't, it's possible to continue working with the wrong object in case we're using a proxy. */
+		if (UNEXPECTED(zend_lazy_object_must_init(obj))) {
+			obj = zend_lazy_object_init(obj);
```

Comment:
`zend_lazy_object_init()` may return `NULL` when initialization fails (in which case `EG(exception)` is set, too).

Unfortunately I don't see other solution than to return `NULL` from `spl_array_get_hash_table_ptr` and to handle that in every caller. (An alternative would be to return a ptr-ptr to an empty array, and to rely on the exception to discard the result of the operation, but then we need to manage that array somehow.)

Reply:
Ouch, okay, I'll do the NULL thing then :/

Reply:
Actually I believe we can just store the empty array in `intern->array`. I pushed a commit that does that, so the total patch stays simple. The only downside is that any further action won't do anything on the object, which may be fine as the object failed initialization anyway. What do you think?

Reply:
Done. I also added some checks to avoid unexpected modification to the sentinel